decorify/profiling.py

```diff
@@ -0,0 +1,123 @@
+""" Module for profiling operations """
+from functools import wraps
+import traceback
+import sys
+from typing import List, Optional
+from decorify.base import decorator
+
+
+def generate_ascii_tree(item, prefix='', is_last=False, is_root=True):
+    """Recursively generates ASCII tree from nested list."""
+    # TODO: can be changed into iterative approach
+    text = ''
+    if isinstance(item, list):
+        node = item[0]
+        children = item[1:]
+    else:
+        node = item
+        children = []
+
+    if is_root:
+        text += node + '\n'
+        child_prefix = ''
+    else:
+        connector = '└── ' if is_last else '├── '
+        text += prefix + connector + node + '\n'
+        child_prefix = prefix + ('    ' if is_last else '│   ')
+
+    last_child_index = len(children) - 1
+    for idx, child in enumerate(children):
+        is_child_last = idx == last_child_index
+        text += generate_ascii_tree(child, child_prefix,
+                                    is_child_last, is_root=False)
+    return text
+
+
+class Tree:
+    """
+    Minimalistic tree structure, created for the purpose of `decorify.crawler`.
+    Each Node containes: name, parent, childrens
+    Methods: to_list - transforms tree structure into nested lists that can be passed to `decorify.profiling.generate_ascii_tree` function.
+    """
+
+    def __init__(self, name: str, parent: Optional["Tree"] = None):
+        self.name = name
+        self.parent = parent
+        self.childrens: List[Tree] = []
+
+    def to_list(self):
+        if not self.childrens:
+            return [self.name]
+        return [self.name, *[child.to_list() for child in self.childrens]]
+
+    def __repr__(self):
+        return generate_ascii_tree(self.to_list())
+
+
+@decorator
+def crawler(build_ins: bool = False, as_list: bool = False, as_tree: bool = True, __func__=None):
```

Comment:
nitpick: It should be "built_ins" instead of "build_ins" 
similar typos in description